packages/babel-generator/src/generators/types.js

```diff
@@ -120,7 +120,11 @@ export function NullLiteral() {
 
 export function NumericLiteral(node: Object) {
   const raw = this.getPossibleRaw(node);
-  const value = node.value + "";
+  const opts = this.format.jsescOption;
+  if (this.format.jsonCompatibleStrings) {
```

Comment:
Why `jsonCompatibleString` should be applied fro `NumericLiteral`? Also `jsonCompatibleString` will be removed in 8.x: https://github.com/babel/babel/pull/9958

Reply:
@JLHwung Tried looking for but did not find what to use instead of `jsonCompatibleString`. That's why used it. What can we use in place of it?

Oh, if it's going to be removed then we will have to replace it in the `StringLiteral` function. Should I make an issue for it?

Reply:
`jsonCompatibleString` is a legacy option and it is equivalent to say `jsescOptions: { json: true }`.

Since it is a legacy option and we never support `jsescOptions` on `NumericLiteral` before, I would say we'd better remove this `jsonCompatibleString` thing and call `jsesc` directly.

```js
const value = jsesc(node.value, opts);
```